### DISCUSSION TITLE
CLI: how tu use parser binary built earlier?

### DISCUSSION THREAD

**Discussion by przepompownia** [tree-sitter/tree-sitter#3336] [Q&A]
For the recent [php_only](https://github.com/tree-sitter/tree-sitter-php/) parser I get crash but only on the binary built by nvim-treesitter (https://github.com/nvim-treesitter/nvim-treesitter/issues/6573). 

I'd like to report the bug but cannot reproduce it using CLI. Using `strace` I found that the CLI uses `~/.cache/tree-sitter/lib/php.so` to parse. It suggests that the parser is built on the fly. I'd like to try the broken binary but not know how to provide the way of building same as nvim-tresitter does (`cc -o parser.so -I./src src/parser.c src/scanner.c -Os -shared -fPIC`) to make a closer reproduction.

**Comment by amaanq** [comment]
move the nvim-treesitter parser in ~/.local/share/nvim/..... to ~/.cache/tree-sitter/lib/php_only.so to make it use that one

Note that php and php_only are two different parsers

**Reply by przepompownia** [reply]
I started testing CLI with `php` because `tree-sitter parse --help` doesn't show how to select `php_only` (tried `--scope php_only` but got `unknown scope`). 

Before creating this question I tried such replacement but got 
```
 $ tree-sitter parse lib/Diff/Tests/RangesForDiffTest.php
Failed to load language for file name RangesForDiffTest.php

Caused by:
    0: Failed to load symbol tree_sitter_php
    1: /home/arctgx/.cache/tree-sitter/lib/php.so: undefined symbol: tree_sitter_php
```

Edit 2. I updated tree-sitter to `0.22.5` (from `0.20.8`) without any noticeable effect.
